### ISSUE TITLE
CoCo_semantic_seg

### ISSUE THREAD

**Issue by dbsdmlgus50** [open-mmlab/mmsegmentation#3423]
I tried to get the coco_semantic_seg data from coco_panoptic and run coco_semantic_seg based on it.

I turned it to the config below and CocoSegDataset brought and used CocoSegDataset provided by mmdetection.

norm_cfg = dict(type='SyncBN', requires_grad=True)
data_preprocessor = dict(
    type='SegDataPreProcessor',
    mean=[123.675, 116.28, 103.53],
    std=[58.395, 57.12, 57.375],
    bgr_to_rgb=True,
    pad_val=0,
    seg_pad_val=255,
    size=(512, 1024))
model = dict(
    type='EncoderDecoder',
    data_preprocessor=dict(
        type='SegDataPreProcessor',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        bgr_to_rgb=True,
        pad_val=0,
        seg_pad_val=255,
        size=(512, 1024)),
    pretrained='open-mmlab://resnet50_v1c',
    backbone=dict(
        type='ResNetV1c',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        dilations=(1, 1, 1, 1),
        strides=(1, 2, 2, 2),
        norm_cfg=dict(type='SyncBN', requires_grad=True),
        norm_eval=False,
        style='pytorch',
        contract_dilation=True),
    neck=dict(
        type='FPN',
        in_channels=[256, 512, 1024, 2048],
        out_channels=256,
        num_outs=4),
    decode_head=dict(
        type='FPNHead',
        in_channels=[256, 256, 256, 256],
        in_index=[0, 1, 2, 3],
        feature_strides=[4, 8, 16, 32],
        channels=128,
        dropout_ratio=0.1,
        num_classes=19,
        norm_cfg=dict(type='SyncBN', requires_grad=True),
        align_corners=False,
        loss_decode=dict(
            type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0)),
    train_cfg=dict(),
    test_cfg=dict(mode='whole'))
dataset_type = 'CocoSegDataset'
crop_size = (512, 1024)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations'),
    dict(
        type='RandomResize',
        scale=(2048, 512),
        ratio_range=(0.5, 2.0),
        keep_ratio=True),
    dict(type='RandomCrop', crop_size=(512, 512), cat_max_ratio=0.75),
    dict(type='RandomFlip', prob=0.5),
    dict(type='PhotoMetricDistortion'),
    dict(type='PackSegInputs')
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='Resize', scale=(2048, 512), keep_ratio=True),
    dict(type='LoadAnnotations'),
    dict(type='PackSegInputs')
]
train_dataloader = dict(
    batch_size=4,
    num_workers=12,
    persistent_workers=True,
    sampler=dict(type='InfiniteSampler', shuffle=True),
    dataset=dict(
        type='CocoSegDataset',
        data_prefix=dict(
            img_path=
            '/workspace/euihyun/mmdetection/data/coco/images/train2017',
            seg_map_path=
            '/workspace/euihyun/mmdetection/data/coco/annotations/panoptic_semseg_train2017'
        ),
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='LoadAnnotations'),
            dict(
                type='RandomResize',
                scale=(2048, 512),
                ratio_range=(0.5, 2.0),
                keep_ratio=True),
            dict(type='RandomCrop', crop_size=(512, 512), cat_max_ratio=0.75),
            dict(type='RandomFlip', prob=0.5),
            dict(type='PhotoMetricDistortion'),
            dict(type='PackSegInputs')
        ]))
val_dataloader = dict(
    batch_size=1,
    num_workers=2,
    persistent_workers=True,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type='CocoSegDataset',
        data_prefix=dict(
            img_path='/workspace/euihyun/mmdetection/data/coco/images/val2017',
            seg_map_path=
            '/workspace/euihyun/mmdetection/data/coco/annotations/panoptic_semseg_val2017'
        ),
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='Resize', scale=(2048, 512), keep_ratio=True),
            dict(type='LoadAnnotations'),
            dict(type='PackSegInputs')
        ]))
test_dataloader = dict(
    batch_size=1,
    num_workers=2,
    persistent_workers=True,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type='CocoSegDataset',
        data_prefix=dict(
            img_path='/workspace/euihyun/mmdetection/data/coco/images/val2017',
            seg_map_path=
            '/workspace/euihyun/mmdetection/data/coco/annotations/panoptic_semseg_val2017'
        ),
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='Resize', scale=(2048, 512), keep_ratio=True),
            dict(type='LoadAnnotations'),
            dict(type='PackSegInputs')
        ]))
val_evaluator = dict(type='IoUMetric', iou_metrics=['mIoU'])
test_evaluator = dict(type='IoUMetric', iou_metrics=['mIoU'])
default_scope = 'mmseg'
env_cfg = dict(
    cudnn_benchmark=True,
    mp_cfg=dict(mp_start_method='fork', opencv_num_threads=0),
    dist_cfg=dict(backend='nccl'))
vis_backends = [dict(type='LocalVisBackend')]
visualizer = dict(
    type='SegLocalVisualizer',
    vis_backends=[dict(type='LocalVisBackend')],
    name='visualizer')
log_processor = dict(by_epoch=False)
log_level = 'INFO'
load_from = None
resume = False
tta_model = dict(type='SegTTAModel')
optimizer = dict(type='SGD', lr=0.01, momentum=0.9, weight_decay=0.0005)
optim_wrapper = dict(
    type='OptimWrapper',
    optimizer=dict(type='SGD', lr=0.01, momentum=0.9, weight_decay=0.0005),
    clip_grad=None)
param_scheduler = [
    dict(
        type='PolyLR',
        eta_min=0.0001,
        power=0.9,
        begin=0,
        end=80000,
        by_epoch=False)
]
train_cfg = dict(type='IterBasedTrainLoop', max_iters=80000, val_interval=8000)
val_cfg = dict(type='ValLoop')
test_cfg = dict(type='TestLoop')
default_hooks = dict(
    timer=dict(type='IterTimerHook'),
    logger=dict(type='LoggerHook', interval=50, log_metric_by_epoch=False),
    param_scheduler=dict(type='ParamSchedulerHook'),
    checkpoint=dict(type='CheckpointHook', by_epoch=False, interval=8000),
    sampler_seed=dict(type='DistSamplerSeedHook'),
    visualization=dict(type='SegVisualizationHook'))
launcher = 'pytorch'
work_dir = './work_dirs/fpn_r50_4xb2-80k_cityscapes-512x1024'

But the results of the study

There are many values of 0 or NaN as shown below What's the problem?

+------------------+-------+-------+
|      Class       |  IoU  |  Acc  |
+------------------+-------+-------+
|      person      | 36.06 | 97.67 |
|     bicycle      | 11.55 |  73.0 |
|       car        |  7.67 | 65.59 |
|    motorcycle    |  1.61 | 86.92 |
|     airplane     |  3.27 | 93.46 |
|       bus        | 30.92 | 74.37 |
|      train       |  9.5  | 73.78 |
|      truck       |  18.8 |  45.9 |
|       boat       | 14.01 | 58.61 |
|  traffic light   |  5.27 | 83.87 |
|   fire hydrant   | 17.08 | 95.58 |
|    stop sign     |  47.8 | 92.33 |
|  parking meter   |  10.8 |  61.5 |
|      bench       |  7.2  | 67.32 |
|       bird       |  6.19 | 83.73 |
|       cat        |  8.21 | 88.02 |
|       dog        | 30.75 | 76.92 |
|      horse       | 17.35 | 80.25 |
|      sheep       | 11.52 | 82.54 |
|       cow        |  0.0  |  0.0  |
|     elephant     |  0.0  |  0.0  |
|       bear       |  0.0  |  0.0  |
|      zebra       |  0.0  |  0.0  |
|     giraffe      |  0.0  |  0.0  |
|     backpack     |  0.0  |  0.0  |
|     umbrella     |  0.0  |  0.0  |
|     handbag      |  0.0  |  0.0  |
|       tie        |  0.0  |  0.0  |
|     suitcase     |  0.0  |  0.0  |
|     frisbee      |  0.0  |  0.0  |
|       skis       |  0.0  |  0.0  |
|    snowboard     |  0.0  |  0.0  |
|   sports ball    |  0.0  |  0.0  |
|       kite       |  0.0  |  0.0  |
|   baseball bat   |  0.0  |  0.0  |
|  baseball glove  |  0.0  |  0.0  |
|    skateboard    |  0.0  |  0.0  |
|    surfboard     |  0.0  |  0.0  |
|  tennis racket   |  0.0  |  0.0  |
|      bottle      |  0.0  |  0.0  |
|    wine glass    |  0.0  |  0.0  |
|       cup        |  0.0  |  0.0  |
|       fork       |  0.0  |  0.0  |
|      knife       |  0.0  |  0.0  |
|      spoon       |  0.0  |  0.0  |
|       bowl       |  0.0  |  0.0  |
|      banana      |  0.0  |  0.0  |
|      apple       |  0.0  |  0.0  |
|     sandwich     |  0.0  |  0.0  |
|      orange      |  0.0  |  0.0  |
|     broccoli     |  0.0  |  0.0  |
|      carrot      |  0.0  |  0.0  |
|     hot dog      |  0.0  |  0.0  |
|      pizza       |  0.0  |  0.0  |
|      donut       |  0.0  |  0.0  |
|       cake       |  0.0  |  0.0  |
|      chair       |  0.0  |  0.0  |
|      couch       |  0.0  |  0.0  |
|   potted plant   |  0.0  |  0.0  |
|       bed        |  0.0  |  0.0  |
|   dining table   |  0.0  |  0.0  |
|      toilet      |  0.0  |  0.0  |
|        tv        |  0.0  |  0.0  |
|      laptop      |  0.0  |  0.0  |
|      mouse       |  0.0  |  0.0  |
|      remote      |  0.0  |  0.0  |
|     keyboard     |  0.0  |  0.0  |
|    cell phone    |  0.0  |  0.0  |
|    microwave     |  0.0  |  0.0  |
|       oven       |  0.0  |  0.0  |
|     toaster      |  0.0  |  0.0  |
|       sink       |  0.0  |  0.0  |
|   refrigerator   |  0.0  |  0.0  |
|       book       |  0.0  |  0.0  |
|      clock       |  0.0  |  0.0  |
|       vase       |  0.0  |  0.0  |
|     scissors     |  0.0  |  0.0  |
|    teddy bear    |  0.0  |  0.0  |
|    hair drier    |  0.0  |  0.0  |
|    toothbrush    |  0.0  |  0.0  |
|      banner      |  0.0  |  0.0  |
|     blanket      |  0.0  |  0.0  |
|      branch      |  0.0  |  0.0  |
|      bridge      |  0.0  |  0.0  |
|  building-other  |  0.0  |  0.0  |
|       bush       |  0.0  |  0.0  |
|     cabinet      |  0.0  |  0.0  |
|       cage       |  0.0  |  0.0  |
|    cardboard     |  0.0  |  0.0  |
|      carpet      |  0.0  |  0.0  |
|  ceiling-other   |  0.0  |  0.0  |
|   ceiling-tile   |  0.0  |  0.0  |
|      cloth       |  0.0  |  0.0  |
|     clothes      |  0.0  |  0.0  |
|      clouds      |  0.0  |  0.0  |
|     counter      |  0.0  |  0.0  |
|     cupboard     |  0.0  |  0.0  |
|     curtain      |  0.0  |  0.0  |
|    desk-stuff    |  0.0  |  0.0  |
|       dirt       |  0.0  |  0.0  |
|    door-stuff    |  0.0  |  0.0  |
|      fence       |  0.0  |  0.0  |
|   floor-marble   |  0.0  |  0.0  |
|   floor-other    |  0.0  |  0.0  |
|   floor-stone    |  0.0  |  0.0  |
|    floor-tile    |  0.0  |  0.0  |
|    floor-wood    |  0.0  |  0.0  |
|      flower      |  0.0  |  0.0  |
|       fog        |  0.0  |  0.0  |
|    food-other    |  0.0  |  0.0  |
|      fruit       |  0.0  |  0.0  |
| furniture-other  |  0.0  |  0.0  |
|      grass       |  0.0  |  0.0  |
|      gravel      |  0.0  |  0.0  |
|   ground-other   |  0.0  |  0.0  |
|       hill       |  0.0  |  0.0  |
|      house       |  0.0  |  0.0  |
|      leaves      |  0.0  |  0.0  |
|      light       |  0.0  |  0.0  |
|       mat        |  0.0  |  0.0  |
|      metal       |  0.0  |  0.0  |
|   mirror-stuff   |  0.0  |  0.0  |
|       moss       |  0.0  |  0.0  |
|     mountain     |  0.0  |  0.0  |
|       mud        |  0.0  |  0.0  |
|      napkin      |  0.0  |  0.0  |
|       net        |  0.0  |  0.0  |
|      paper       |  0.0  |  0.0  |
|     pavement     |  0.0  |  0.0  |
|      pillow      |  0.0  |  0.0  |
|   plant-other    |  0.0  |  0.0  |
|     plastic      |  0.0  |  0.0  |
|     platform     |  0.0  |  0.0  |
|   playingfield   |  nan  |  nan  |
|     railing      |  nan  |  nan  |
|     railroad     |  nan  |  nan  |
|      river       |  nan  |  nan  |
|       road       |  nan  |  nan  |
|       rock       |  nan  |  nan  |
|       roof       |  nan  |  nan  |
|       rug        |  nan  |  nan  |
|      salad       |  nan  |  nan  |
|       sand       |  nan  |  nan  |
|       sea        |  nan  |  nan  |
|      shelf       |  nan  |  nan  |
|    sky-other     |  nan  |  nan  |
|    skyscraper    |  nan  |  nan  |
|       snow       |  nan  |  nan  |
|   solid-other    |  nan  |  nan  |
|      stairs      |  nan  |  nan  |
|      stone       |  nan  |  nan  |
|      straw       |  nan  |  nan  |
| structural-other |  nan  |  nan  |
|      table       |  nan  |  nan  |
|       tent       |  nan  |  nan  |
|  textile-other   |  nan  |  nan  |
|      towel       |  nan  |  nan  |
|       tree       |  nan  |  nan  |
|    vegetable     |  nan  |  nan  |
|    wall-brick    |  nan  |  nan  |
|  wall-concrete   |  nan  |  nan  |
|    wall-other    |  nan  |  nan  |
|    wall-panel    |  nan  |  nan  |
|    wall-stone    |  nan  |  nan  |
|    wall-tile     |  nan  |  nan  |
|    wall-wood     |  nan  |  nan  |
|   water-other    |  nan  |  nan  |
|    waterdrops    |  nan  |  nan  |
|   window-blind   |  nan  |  nan  |
|   window-other   |  nan  |  nan  |
|       wood       |  nan  |  nan  |
+------------------+-------+-------+

Thank you for reading
